### PR TITLE
Add YOLO training example with Roboflow dataset integration

### PR DESCRIPTION
**Pull request by Panupong-benz**

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Panupong-benz, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review readiness, and an engineer will assist you soon. Since this PR adds a new Python training example for YOLO + Roboflow integration, please review the checklist below to help ensure a smooth merge. 😊

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: examples/roboflow_yolo_training.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+model = YOLO("yolo11n.pt")  # โหลด pretrained model
+
+results = model.train(
+    data=DATA_YAML,
+    epochs=50,          # จำนวนรอบการเทรน (ปรับได้ 50-300)
+    imgsz=640,          # ขนาดรูป input
+    batch=16,           # batch size (ลดลงถ้า GPU memory ไม่พอ)
+    name="roboflow_assignment",   # ชื่อ run
+    patience=10,        # หยุดเร็วถ้าไม่ดีขึ้นใน 10 epoch
+    device=0,           # 0 = GPU, 'cpu' = CPU
```
Comment (UltralyticsAssistant) [inline examples/roboflow_yolo_training.py:66]: ⚠️ **HIGH**: Hardcoding `device=0` makes this example fail immediately on any machine without a CUDA GPU. Since example scripts are often run on CPU-only laptops or CI, this should either let Ultralytics auto-select the device or branch on CUDA availability before training.

File: examples/roboflow_yolo_training.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+print("\n========== PERFORMANCE METRICS ==========")
+print(f"mAP50        : {metrics.box.map50:.4f}")   # mAP @ IoU=0.50
+print(f"mAP50-95     : {metrics.box.map:.4f}")     # mAP @ IoU=0.50:0.95
+print(f"Precision    : {metrics.box.mp:.4f}")
+print(f"Recall       : {metrics.box.mr:.4f}")
+print("=========================================")
+
+# ดู per-class metrics
+print("\n--- Per-Class AP50 ---")
+for i, cls_name in enumerate(metrics.names.values()):
```
Comment (UltralyticsAssistant) [inline examples/roboflow_yolo_training.py:90]: 💡 **MEDIUM**: This per-class reporting can mislabel AP50 values. `metrics.box.ap50` is ordered by `metrics.box.ap_class_index` and only includes classes present in the validation labels, so indexing it with the raw class loop can print another class's AP when some classes are missing from `val`. Use `ap_class_index` or `metrics.box.maps` to map scores back to class ids correctly.

File: examples/roboflow_yolo_training.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+# ============================================================
+# STEP 5: Inference Example
+# ============================================================
+# โหลด best model ที่เทรนได้
+
+best_model_path = os.path.join(str(results.save_dir), "weights", "best.pt")
+best_model = YOLO(best_model_path)
+
+# --- Inference บนรูปเดียว ---
+IMAGE_PATH = "path/to/your/test/image.jpg"   # <-- แก้ตรงนี้
```
Comment (UltralyticsAssistant) [inline examples/roboflow_yolo_training.py:104]: 💡 **MEDIUM**: This inference block is active even though `IMAGE_PATH` is still a placeholder, so the script will crash after a successful train/val run unless the user edits this line. For a repo example, it's safer to guard this with an existence check or comment it out like the folder/video examples below.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Nice end-to-end example overall, but there are a few usability/correctness issues to address before merging: the training device is hardcoded to CUDA, the per-class AP50 loop can report incorrect labels, and the active placeholder inference path makes the script fail unless manually edited.

💬 Posted 3 inline comments

**Comment by glenn-jocher** [comment]
Thanks for the update — since this PR has been closed, we’ll treat it as complete; if you’d like to revisit the Roboflow training example later, please open a new minimal PR scoped to that single documentation/example change.
